adjust/kpis_test.go

```diff
@@ -51,16 +26,30 @@ func TestKPIService_ListWithOptions(t *testing.T) {
 		Sandbox:           false,
 		HumanReadableKpis: true,
 		Grouping:          "day,networks,campaigns,adgroups,creatives",
-		Reattributed:      "all",
-	}
-	got, _, err := client.KPI.List(context.Background(), &opt)
-	if err != nil {
-		t.Errorf("KPI.List returned error: %v", err)
-	}
+		Reattributed:      "all"},
+	},
+}
 
-	want := &KPI{}
-	responseToInterface(loadFixture("kpis_options.json"), &want)
-	if !reflect.DeepEqual(got, want) {
-		t.Errorf("KPI.List = %+v, want %+v", got, want)
+func TestKPIService_List(t *testing.T) {
+	for _, tt := range kpiTest {
+		t.Run(tt.name, func(t *testing.T) {
+			client, mux, _, teardown := setup()
+			t.Log(fmt.Sprintf("Setup Done %s", tt.name))
+			defer teardown()
+			mux.HandleFunc(tt.path, func(w http.ResponseWriter, r *http.Request) {
+				testMethod(t, r, "GET")
+				w.WriteHeader(http.StatusOK)
+				w.Write(loadFixture(tt.fixture))
+			})
+			got, _, err := client.KPI.List(context.Background(), tt.opts)
+			if err != nil {
+				t.Errorf("KPI.List returned error: %v", err)
+			}
+			want := &KPI{}
+			responseToInterface(loadFixture(tt.fixture), &want)
+			if !reflect.DeepEqual(got, want) {
+				t.Errorf("KPI.List = %+v, want %+v", got, want)
+			}
+		})
 	}
 }
```

Comment:
So in general the code looks very good to me. I couldn't find any obvious mistakes and I like the way the table test was written. Only slight changes for better readability, where i added comments.